File: Core/Domain/BaseTriangle.cs
namespace Core.Domain

{
    public abstract class BaseTriangle
    {
        public int Id { get; set; }
        public int V1 { get; set; }   // each vertex is a node
        public int V2 { get; set; }
        public int V3 { get; set; }
        public double AvgX { get; set; }  // avgx and avgy are x and y coords of triangle center
        public double AvgY { get; set; }     
        public Edge Edge1 { get; set; }    //holds information related to edge opposide V1
        public Edge Edge2 { get; set; }    //holds information related to edge opposide V2
        public Edge Edge3 { get; set; }    //holds information related to edge opposide V3
        public double Flux { get; set; }  // total flux for the element
        public bool Complete { get; set; }
        public double Area { get; }
    }
}

File: Mesh/Data/Domain/BaseTriangle.vb
﻿Namespace Models
    Public MustInherit Class BaseTriangle

#Region "public properties"
        Public Property Id As Integer
        Public Property V1 As Integer   'each vertex is a node
        Public Property V2 As Integer
        Public Property V3 As Integer
        Public Property AvgX As Double  'avgx and avgy are x and y coords of triangle center
        Public Property AvgY As Double
        Public Property L1 As Double      'length of side opposite v1
        Public Property L2 As Double
        Public Property L3 As Double
        Public Property S1 As String       'used to flag if surface, boundary or interior side
        Public Property S2 As String
        Public Property S3 As String
        Public Property X1 As Double
        Public Property Y1 As Double
        Public Property X2 As Double
        Public Property Y2 As Double
        Public Property X3 As Double
        Public Property Y3 As Double

        Public Property theta1 As Double
        Public Property theta2 As Double
        Public Property theta3 As Double

        Public Property Complete As Boolean
        Public ReadOnly Property Area As Double
#End Region

    End Class
End Namespace
